raiden/utils/http.py

```diff
@@ -45,6 +47,12 @@ def start(self):
                 stdin = stdout = stderr = self.stdio
             env = os.environ.copy()
             env[ENV_UUID] = self._uuid
+
+            executable = shlex.split(command)[0]
```

Comment:
Hmm, but what if the user pass a variable to the script, like  `--local-matrix="STDOUT_SYNAPSE=1 ./.synapse/run_synapse.sh > my.log"`? I think the safest way will be handling `CalledProcessError` exception and checking `returncode == 127`, which is for `No such file or directory` system errors

Reply:
`STDOUT_SYNAPSE=1 raiden smoketest --local-matrix="./.synapse/run_synapse.sh > my.log"` should do the trick here, since the code is actually copying the environment and passing it to `subprocess.Popen()`.
I prefer this solution, since it allows to fail fast in `smoketest`.

Don't you think it is flexible enough?

Reply:
I don't think it's flexible enough. Yes, on my example, the variable could be passed in the raiden environment, but that could have side effects, and we couldn't think of every case an specific parsing like that could go wrong. OTOH, I don't think the difference between `split + which` vs `subprocess.Popen + except` would be noticeable, `Popen` will execute asap, so the difference would be negligible for a smoketest which doesn't run that often (twice per test run, travis takes minutes to set-up the environment to run each) to be worth optimizing a few `ms`.